cmake/generic_toolchain.cmake

```diff
@@ -4,10 +4,14 @@ if(NOT TOOLCHAIN_ROOT)
   if(DEFINED ENV{TOOLCHAIN_ROOT})
     # Support for out-of-tree toolchain
     set(TOOLCHAIN_ROOT $ENV{TOOLCHAIN_ROOT})
+    set(TOOLCHAIN_USE_CUSTOM 1)
```

Comment:
at line 40, we include `generic_toolchain.cmake` based on `TOOLCHAIN_ROOT`
https://github.com/zephyrproject-rtos/zephyr/blob/1f26ce0b872c07222fd87119af76ac0b381b8882/cmake/generic_toolchain.cmake#L40

Isn't it better to ask users having a custom toolchain to place this line in there instead, like ?
```
# ${TOOLCHAIN_ROOT}/cmake/toolchain/${ZEPHYR_TOOLCHAIN_VARIANT}/generic.cmake
set(TOOLCHAIN_USE_CUSTOM 1)

....
```

That will allow users with a custom toolchain based on, e.g. LLVM, to still use `<toolchain/llvm.h>` if that actually work in their setup.
In such a case they can simply decide not to have `set(TOOLCHAIN_USE_CUSTOM 1)` in their generic.cmake.

Reply:
@tejlmand Thanks for your feedback and for the suggestion. 
I agree that having the users of off-tree toolchains decide whether they want the other.h included, instead of having that decided internally in the Zephyr cmake system, is the more flexible and thereby the better solution for that.
I will rework the commit to expect that.